javaslang/src/main/java/javaslang/collection/RedBlackTree.java

```diff
@@ -68,7 +68,7 @@
 
     static <T extends Comparable<? super T>> RedBlackTree<T> ofAll(Iterable<? extends T> values) {
         Objects.requireNonNull(values, "values is null");
-        return ofAll((Comparator<? super T> & Serializable) T::compareTo, values);
+        return ofAll(Comparators.naturalComparator(), values);
```

Comment:
Mmmh, why that? Does the following work?

```java
return ofAll((Comparator<T> & Serializable) T::compareTo, values);
```

Widening the type is safe in this case.